plugins/meshy/meshy.js

```diff
@@ -0,0 +1,1236 @@
+        }
+        mesh.addTo(group).init();
+    }
+}
+
+function uvOnSave(...uv) { 
+    uv[1] = Project.texture_height - uv[1] //Invert y axis
+    if (!settings["normalized_uvs"].value) return uv
+    uv[0] /= Project.texture_width
+    uv[1] /= Project.texture_height
+    return uv
+}
+
+//gets vertices of a Mesh and applys transformations to the points so that they can be exported
+function getVertices(mesh) {
+	const verts = Object.entries(mesh.vertices).map( ( [key, point ]) => {
+		point = rotatePoint(point, mesh.origin, mesh.rotation)
+        point.V3_add(mesh.position[0], mesh.position[1], mesh.position[2])
+		return [ key, point ]
+	}) 
+	return verts;
+}
+
+/**
+ * Gets the vertex normal of a mesh
+ * @param {Mesh} mesh The mesh to get the vertex normal from
+ * @param {string} vertexKey The key of the vertex
+ * @param {Map} vertexFacesMap The map of vertex faces
+ * The vertexFacesMap is used to get the faces of the vertex 
+ * This so we don't have to loop through the faces for each vertex
+ */
+function getVertexNormal(mesh, vertexKey, vertexFacesMap) {
+    if (settings["skip_normals"].value) return [ 0,1,0 ];
+    let normalSum = [0, 0, 0];
+    let faceCount = 0;
+
+    const faces = vertexFacesMap.get(vertexKey) || []
+    for (let face of faces) {
+        face = mesh.faces[face];
+        let faceNormal = face.getNormal();
+        normalSum[0] += faceNormal[0];
+        normalSum[1] += faceNormal[1];
+        normalSum[2] += faceNormal[2];
+        faceCount++;
+    }
+
+    let normalLength = Math.sqrt(normalSum[0] * normalSum[0] + normalSum[1] * normalSum[1] + normalSum[2] * normalSum[2]);
+    if (normalLength === 0) {
+        return [0, 1, 0]; // Default to up vector if normal is zero
+    }
+    return [
+        normalSum[0] / normalLength,
+        normalSum[1] / normalLength,
+        normalSum[2] / normalLength
+    ];
+}
+
+function multiplyScalar(vec, scalar) {
+    return vec.map((coord) => coord * scalar);
+}
+
+function rotatePoint(point, center, rotation) {
+    // Convert rotation angles to radians
+    const [rx, ry, rz] = rotation.map(Math.degToRad);
+
+    // Translate point to origin
+    let [x, y, z] = point.map((coord, i) => coord - center[i]);
+
+    // Rotate around X-axis
+    let temp = y;
+    y = y * Math.cos(rx) - z * Math.sin(rx);
+    z = temp * Math.sin(rx) + z * Math.cos(rx);
+
+    // Rotate around Y-axis
+    temp = x;
+    x = x * Math.cos(ry) + z * Math.sin(ry);
+    z = -temp * Math.sin(ry) + z * Math.cos(ry);
+
+    // Rotate around Z-axis
+    temp = x;
+    x = x * Math.cos(rz) - y * Math.sin(rz);
+    y = temp * Math.sin(rz) + y * Math.cos(rz);
+
+    // Translate back
+    return [
+        x + center[0],
+        y + center[1],
+        z + center[2]
+    ];
+}
+(function() {
+const pluginInfo = {"name":"Meshy","id":"meshy","version":"1.0.2","repository":"https://github.com/Shadowkitten47/Meshy"};
+
+if (!settings["normalized_uvs"])
```

Comment:
Settings IDs are very generic right now. These should be namespaced (e.g. meshy_normalized_uvs) to avoid conflicts with other plugins. The same applies to names. When looking through the settings, users will have no idea what "Meta Data" refers to without context.